packages/jsx-directive/src/core/index.ts

```diff
@@ -35,8 +48,45 @@ export function transformJsxDirective(code: string, id: string) {
 
   const s = new MagicString(code)
   for (const { ast, offset } of asts) {
-    vIfTransform(ast, s, offset)
-    vForTransform(ast, s, offset)
+    if (!REGEX_VUE_DIRECTIVE.test(s.sliceNode(ast, { offset }))) return
+
+    const vForNodes: JsxDirectiveNode[] = []
+    const vIfMap = new Map<Node, JsxDirectiveNode[]>()
+    walkAST<Node>(ast, {
+      enter(node, parent) {
+        if (node.type !== 'JSXElement') return
+
+        let vIfAttribute
+        let vForAttribute
+        for (const attribute of node.openingElement.attributes) {
+          if (attribute.type !== 'JSXAttribute') continue
+          if (
+            ['v-if', 'v-else-if', 'v-else'].includes(`${attribute.name.name}`)
+          )
+            vIfAttribute = attribute
+          if (attribute.name.name === 'v-for') vForAttribute = attribute
+        }
+
+        if (vIfAttribute) {
+          if (!vIfMap.has(parent!)) vIfMap.set(parent!, [])
+          vIfMap.get(parent!)?.push({
+            node,
+            attribute: vIfAttribute,
+            parent,
+          })
+        }
+        if (vForAttribute) {
+          vForNodes.push({
+            node,
+            attribute: vForAttribute,
+            parent: vIfAttribute ? undefined : parent,
+          })
+        }
+      },
+    })
+
+    transformVIf([...vIfMap.values()].flat(), s, offset)
```

Comment:
```suggestion
    transformVIf(Array.from(vIfMap.values()).flat(), s, offset)
```

Reply:
Ok, I changed it. But I want to know how they are different, performance?

Reply:
For more readability